ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/example/CommonUtils.java

```diff
@@ -0,0 +1,190 @@
+import io.ballerina.compiler.api.symbols.TypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.values.ConstantValue;
+import io.ballerina.openapi.service.mapper.diagnostic.OpenAPIMapperDiagnostic;
+import io.ballerina.openapi.service.mapper.example.field.RecordFieldExampleMapper;
+import io.swagger.v3.oas.models.examples.Example;
+import io.swagger.v3.oas.models.media.ObjectSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import org.wso2.ballerinalang.compiler.tree.BLangConstantValue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.ballerina.openapi.service.mapper.Constants.BALLERINA;
+import static io.ballerina.openapi.service.mapper.Constants.EMPTY;
+import static io.ballerina.openapi.service.mapper.Constants.OPENAPI;
+
+/**
+ * This {@link CommonUtils} class represents the common utility functions of example mapper.
+ *
+ * @since 2.1.0
+ */
+public final class CommonUtils {
+
+    private static final String EXAMPLE_VALUES = "ExampleValues";
+    private static final String EXAMPLE_VALUE = "ExampleValue";
+    private static final String VALUE = "value";
+    private static final String QUOTE = "\"";
+    private static final String COMMA = ",";
+    private static final String BRACKET_START = "[";
+    private static final String BRACKET_END = "]";
+    private static final String COLON = ":";
+    private static final String BRACE_START = "{";
+    private static final String BRACE_END = "}";
+    public static final String INLINE_REC = "<inline-record>";
+
+    private CommonUtils() {
+    }
+
+    public static Optional<Object> extractOpenApiExampleValue(List<AnnotationAttachmentSymbol> annotations,
+                                                              SemanticModel semanticModel)
+            throws JsonProcessingException {
+        Optional<AnnotationAttachmentSymbol> openAPIExample = annotations.stream()
+                .filter(annotAttachment -> isOpenAPIExampleAnnotation(annotAttachment, semanticModel))
+                .findFirst();
+        if (openAPIExample.isEmpty() || !openAPIExample.get().isConstAnnotation() ||
+                openAPIExample.get().attachmentValue().isEmpty()) {
+            return Optional.empty();
+        }
+
+        Object exampleValue = openAPIExample.get().attachmentValue().get().value();
+        return getExampleFromValue(exampleValue);
+    }
+
+    private static Optional<Object> getExampleFromValue(Object exampleValue) throws JsonProcessingException {
+        if (exampleValue instanceof ConstantValue constantExampleValue) {
+            exampleValue = constantExampleValue.value();
+        } else if (exampleValue instanceof BLangConstantValue constantExampleValue) {
+            exampleValue = constantExampleValue.value;
```

Comment:
```suggestion
            exampleValue = constantExampleValue.value();
```

Reply:
There is no `value()` method on `BLangConstantValue`. Actually there is a bug in lang which exposes this type: https://github.com/ballerina-platform/ballerina-lang/issues/43115. Once that is fixed, all the constant values should be of type `ConstantValue`. Then we can remove this if check with `BLangConstantValue`.